modules/tinymce/src/core/test/ts/browser/lookup/UserLookupTest.ts

```diff
@@ -0,0 +1,345 @@
+import { describe, it } from '@ephox/bedrock-client';
+import { Arr, Fun } from '@ephox/katamari';
+import { TinyHooks } from '@ephox/wrap-mcagar';
+import * as Chai from 'chai';
+import { expect } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+
+import type Editor from 'tinymce/core/api/Editor';
+import type { User } from 'tinymce/core/lookup/UserLookup';
+
+Chai.use(chaiAsPromised);
+
+const createMockUser = (id: string): User => ({
+  id,
+  name: 'Test User',
+  avatar: 'test-avatar.png',
+  description: 'Test Description'
+});
+
+describe('browser.tinymce.core.UserLookupTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    base_url: '/project/tinymce/js/tinymce',
+    user_id: 'test-user-1',
+    fetch_users: (userIds: string[]): Promise<User[]> => {
+      return new Promise((resolve) => {
+        setTimeout(() => {
+          const validIds = Arr.filter(userIds, (id): id is string =>
+            typeof id === 'string' && id.length > 0
+          );
+
+          // Only create users for valid test IDs
+          const users = Arr.bind(validIds, (id) =>
+            id.startsWith('test-user-') ? [ createMockUser(id) ] : []
+          );
+
+          resolve(users);
+        }, 0);
+      });
+    },
+  });
+
+  it('TINY-11974: Should return the configured user ID', () => {
+    const editor = hook.editor();
+    const currentUserId = editor.userLookup.getUserId();
+    expect(currentUserId).to.equal('test-user-1', 'Should return the configured user ID');
+  });
+
+  it('TINY-11974: Should have frozen getUserId but mutable fetchUsers', () => {
+    const editor = hook.editor();
+    const lookup = editor.userLookup;
+
+    try {
+      lookup.getUserId = Fun.constant('new-id');
+      expect.fail('Should not allow modifying getUserId');
+    } catch (error) {
+      expect(error).to.be.instanceOf(Error, 'Should throw error when trying to modify read-only function');
+    }
+
+    expect(lookup.getUserId()).to.equal('test-user-1', 'Should maintain original value');
+  });
+
+  it('TINY-11974: Should fetch users and return array of promises', () => {
+    const editor = hook.editor();
+    const userIds = [ 'test-user-1' ];
+
+    const promises = editor.userLookup.fetchUsers(userIds);
+
+    expect(promises).to.be.an('array', 'Should return an array of promises');
+    expect(promises).to.have.lengthOf(1, 'Should return array with one promise');
+    expect(promises[0]).to.be.instanceOf(Promise, 'Should return a Promise');
```

Comment:
Testing the type checker is generally pointless. Unless someone edits the API type signature, which would be very obvious in a PR, the array and promise type checks can't fail (or the test wouldn't compile).